### ISSUE TITLE
Compare validator stop working in Release

### ISSUE THREAD

**Issue by magicznyadrian** [joanpablo/reactive_forms#159]
I have a problem with the code below. In debug mode everything works, but when I run the application via flutter run --release validatior "compare" stops working (no error message, but in form.errors: ' form invalid{compare: { control: 50, compareControl: 10, option: CompareOption.greaterOrEqua}). 
Validatior "required" still work. 
The problem is occurring on Android and ios.

```
FormGroup get form => fb.group({
        'value1': 10,
        'value2': [0, Validators.required]
      }, [
        Validators.compare(
            'value1', 'value2', CompareOption.greaterOrEqual),
      ]);

@override
Widget build(BuildContext context) {
  ReactiveFormBuilder(
          form: () => form,
          builder: (context, form, child) {

...

     ReactiveTextField<int>(
              keyboardType: TextInputType.number,
              formControlName: 'value2',
              validationMessages: (_) => {
                    ValidationMessage.required: 'equired'
                    ValidationMessage.compare: 'greaterOrEqual'})

})
```

Flutter doctor output:

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.1, on macOS 11.3 20E232 darwin-x64, locale pl-PL)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.56.0)
[✓] Connected device (3 available)

• No issues found!
```

**Comment by joanpablo** [comment]
Thanks @magicznyadrian,

I will take a look and let you know.

Best regards

**Comment by joanpablo** [comment]
Hi @magicznyadrian,

Sorry for the late answer. But looking at your example: Are you trying to compare a **String** value with an **int** value?

**Comment by magicznyadrian** [comment]
Thanks @joanpablofor your quick reply,

Ofcourse no, sorry for the mistake in my example. I fixed it.

**Comment by joanpablo** [comment]
Hi @magicznyadrian,

This form definition:

```dart
form => fb.group({
        'value1': 10,
        'value2': [0, Validators.required]
      }, [
        Validators.compare('value1', 'value2', CompareOption.greaterOrEqual),
      ]);
```
Defines that the **value1** (10) is **greater or equals** than **value2** (0) which is true. The form is **valid** and also the controls.

**Comment by magicznyadrian** [comment]
The problem is that with the same data in debug mode it works fine but in release(flutter run --release) it doesn't:
In release:
when form valid 
-form.valid return false, no error mesasge
when form invalid:
-form.valid return false, no error mesasge

**Comment by joanpablo** [comment]
mmm, interesting. The tests all passed successfully. Let me create an example and take a look at this issue. Thanks.

**Comment by joanpablo** [comment]
Hi @magicznyadrian,

I have released v10.4.0 fixing the issue. Please test it and let me know if something is missing.

Best regards and thanks once again for the issue.

**Comment by magicznyadrian** [comment]
Works perfectly! Thank you.
